tests/longhaul/base_object_test.go

```diff
@@ -0,0 +1,105 @@
+package longhaul
+
+import (
+	"math/rand"
+	"sync"
+	"testing"
+
+	"github.com/icrowley/fake"
+	"github.com/rook/rook/tests/framework/clients"
+	"github.com/rook/rook/tests/framework/installer"
+	"github.com/rook/rook/tests/framework/utils"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"time"
+)
+
+// Set up rook cluster if necessary
+// create object store if necessary
+// create object store user if necessary
+// returns k8s helper, install helper and s3 helper for the object store that was just created.
+func setUpRook(t func() *testing.T, namespace string) (*utils.K8sHelper, *installer.InstallHelper, *clients.TestClient) {
```

Comment:
isn't there a generic helper that sets up rook that can be used across integration and longhaul tests? this seems 
 like a duplicate.

Reply:
There is a generic Install Rook which all the tests use, this is a common method for longhaul. I have another PR where we created a interfae for Teardown - uninstalling rook. I was planning to adding set up to the interface as well. I favor of keeping this PR small, I am going to do that saperately, as all integration tests have to be updated as well.

Reply:
ok, i'd recommend we get this refactoring done very soon. These duplications keep coming up.

Reply:
Yes, i have created a ticket: https://github.com/rook/rook/issues/1238.  I work on as soon as PR https://github.com/rook/rook/pull/1223 and this one is merged.